upup/pkg/fi/cloudup/awstasks/iamrolepolicy.go

```diff
@@ -192,12 +280,33 @@ func (e *IAMRolePolicy) policyDocumentString() (string, error) {
 }
 
 type terraformIAMRolePolicy struct {
-	Name           *string            `json:"name"`
+	Name           *string            `json:"name,omitempty"`
 	Role           *terraform.Literal `json:"role"`
-	PolicyDocument *terraform.Literal `json:"policy"`
+	PolicyDocument *terraform.Literal `json:"policy,omitempty"`
+	PolicyArn      *string            `json:"policy_arn,omitempty"`
```

Comment:
I'm staring to wonder if we'd be better off with an entirely different type for the different terraform resource.

```go
type terraformIAMRolePolicyAttachment struct {
	Role      *terraform.Literal `json:"role"`
	PolicyArn *string            `json:"policy_arn,omitempty"`
}
```

We could probably achieve that in a followup PR though.

Reply:
Yeah, I had made an earlier attempt at separating these out unsuccessfully.